src/profile-logic/call-tree.js

```diff
@@ -300,10 +305,14 @@ function _getInvertedStackSelfTimes(
     callNodeIndex++
   ) {
     const prefixCallNode = callNodeTable.prefix[callNodeIndex];
-    if (prefixCallNode !== -1) {
-      callNodeToRoot[callNodeIndex] = callNodeToRoot[prefixCallNode];
-    } else {
+    if (prefixCallNode === -1) {
+      // callNodeIndex is a root node
       callNodeToRoot[callNodeIndex] = callNodeIndex;
+    } else {
+      // callNodeIndex isn't a root node. But because how callNodeIndex is
+      // built, we know that we already visited its root node and can just reuse
```

Comment:
It needs to be "already visited its *parent* node" instead of "root".

How about: "The callNodeTable guarantees that a callNode's prefix always comes before the callNode; prefix references are always to lower callNode indexes and never to higher indexes. We are iterating the callNodeTable in forwards direction (starting at index 0) so we know that we have already visited the current call node's prefix call node and can reuse its stored root node."

Reply:
I took your sentence and added at the end `Recursively we know the prefix' stored root node is the node value we're looking for.`